be/src/exec/vectorized/json_scanner.cpp

```diff
@@ -715,80 +735,209 @@ void JsonReader::_construct_column(simdjson::ondemand::value& value, Column* col
+    simdjson::ondemand::number_type tp;
+
+    auto err = value.get_number_type().get(tp);
+    if (UNLIKELY(err)) {
+        column->append_nulls(1);
+        return;
+    }
+
+    if (tp == simdjson::ondemand::number_type::signed_integer) {
+        int64_t i64 = 0;
+        err = value.get_int64().get(i64);
         if (UNLIKELY(err)) {
             column->append_nulls(1);
-            break;
+            return;
         }
 
-        switch (tp) {
-        case simdjson::ondemand::number_type::signed_integer: {
-            int64_t i64;
-            err = value.get_int64().get(i64);
-            if (UNLIKELY(err)) {
-                column->append_nulls(1);
-                break;
-            }
+        switch (type_desc.type) {
+        case TYPE_BIGINT: {
+            column->append_numbers(&i64, sizeof(i64));
+            break;
+        }
 
-            auto f = fmt::format_int(i64);
-            column->append_strings(std::vector<Slice>{Slice(f.data(), f.size())});
+        case TYPE_INT: {
+            auto i32 = static_cast<int32_t>(i64 & 0xffffffff);
+            column->append_numbers(&i32, sizeof(i32));
             break;
         }
 
-        case simdjson::ondemand::number_type::unsigned_integer: {
-            uint64_t u64;
-            err = value.get_uint64().get(u64);
-            if (UNLIKELY(err)) {
-                column->append_nulls(1);
-                break;
-            }
+        case TYPE_SMALLINT: {
+            auto i16 = static_cast<int16_t>(i64 & 0xffff);
+            column->append_numbers(&i16, sizeof(i16));
+            break;
+        }
 
-            auto f = fmt::format_int(u64);
-            column->append_strings(std::vector<Slice>{Slice(f.data(), f.size())});
+        case TYPE_TINYINT: {
+            auto i8 = static_cast<int8_t>(i64 & 0xff);
+            column->append_numbers(&i8, sizeof(i8));
             break;
         }
 
-        case simdjson::ondemand::number_type::floating_point_number: {
-            std::string_view sv;
-            err = simdjson::to_json_string(value).get(sv);
-            if (UNLIKELY(err)) {
-                column->append_nulls(1);
-                break;
-            }
+        // Float column, integer value.
+        case TYPE_FLOAT: {
+            auto f = static_cast<float>(i64);
+            column->append_numbers(&f, sizeof(f));
+            break;
+        }
 
-            column->append_strings(std::vector<Slice>{Slice{sv.data(), sv.size()}});
+        // Double column, integer value.
+        case TYPE_DOUBLE: {
+            auto d = static_cast<double>(i64);
+            column->append_numbers(&d, sizeof(d));
             break;
         }
 
         default: {
             column->append_nulls(1);
+            return;
+        }
+        }
+
+    } else if (tp == simdjson::ondemand::number_type::floating_point_number) {
+        double d = 0;
+        err = value.get_double().get(d);
+        if (UNLIKELY(err)) {
+            column->append_nulls(1);
```

Comment:
Please check the return value to ensure that the column is indeed a nullable column.

Reply:
The column is already converted to `NullableColumn`.